### DISCUSSION TITLE
proj.db and vcpkg

### DISCUSSION THREAD

**Discussion by timoore** [pelicanmapping/rocky#20] [Q&A / Support] [answered]
Hi,
This isn't actually a rocky question, but I stole a bunch of the SRS code for [vsgCs](https://github.com/timoore/vsgCs)  and thought that I'd ask the experts. I install PROJ using vcpkg. The code works well, but it can't find proj.db unless I set the `PROJ_DATA` environment variable. Have you run into this problem before? I'm wondering if something is wrong with my vcpkg setup, or perhaps I can pass an option to vcpkg when PROJ is installed.

Thanks,
Tim

**Comment by gwaldron** [comment]
If you set `PROJ_DEBUG=2` it will tell you where PROJ is looking for its db file.
A lot of times it will still actually find it even though rocky spits out the warning message about the envvar not being set.

**Comment by timoore** [comment]
Without specifying PROJ_DATA, it says that it's tryling to open "proj.db".
Oh well, thanks for the debugging pointer!

**Reply by gwaldron** [reply]
On Windows, if I leave PROJ_DATA unset, I get this:
```
[rocky info] PROJ says: pj_open_lib(proj.ini): call fopen(H:\devel\rocky\build\vcpkg_installed\x64-windows/share/proj/proj.ini) - succeeded
[rocky info] PROJ says: pj_open_lib(proj.db): call fopen(H:\devel\rocky\build\vcpkg_installed\x64-windows/share/proj/proj.db) - succeeded
[rocky warning] Environment variable PROJ_DATA is not set
```
where `H:\devel\rocky\build\vcpkg_installed\x64-windows` contains my vcpkg dependencies.

This works because of this code:
https://github.com/OSGeo/PROJ/blob/657c69c6937c8f7601daf995a0f75207ecaa4fc1/src/filemanager.cpp#L1508

Looks like PROJ also have a user-settable "file_finder" that we could use to locate proj.db. Lots of options, thoughts appreciated.

**Comment by gwaldron** [comment]
@timoore did you get this working?

**Reply by timoore** [reply, marked as answer]
Nope, I've gotten by with setting PROJ_DATA. I haven't really had time to figure out what might be going wrong in my vcpkg installation.
